plugins/modules/xen_orchestra.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021 Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: xen_orchestra
+    short_description: Management of instances on Xen Orchestra 
+    version_added: 4.1.0
+    author:
+      - Samori Gorse (@shinuza) <samorigorse@gmail.com>
+    requirements:
+      - websocket-client >= 1.0.0
+    description:
+      - Allows you to create/delete/restart/stop instances on Xen Orchestra
+    options:
+      api_host:
+        description: API host to XOA API.
+        type: str
+      user:
+        description: Xen Orchestra user.
+        required: true
+        type: str
+      password:
+        description: Xen Orchestra password.
+        required: true
+        type: str
+      validate_certs:
+        description: Verify TLS certificate if using HTTPS.
+        type: boolean
+        default: true
+      use_ssl:
+        description: Use wss when connecting to the Xen Orchestra API
+        type: boolean
+        default: true
+      state:
+        description: State in which the Virtual Machine should be
+        choices: ['present', 'started', 'absent', 'stopped', 'restarted']
+        default: present
+      label:
+        description: Label of the Virtual Machine to create, can be used when O(state=present)
+        type: boolean
```

Comment:
I don't think the long variants are allowed for modules so far.
```suggestion
        type: bool
```